ci-operator/step-registry/openshift/image-based/install/openshift-image-based-install-commands.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+export PS4='+ $(date "+%T.%N") \011'
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-privatekey")
+
+LCA_PULL_SECRET_FILE="/var/run/pull-secret/.dockerconfigjson"
+CLUSTER_PULL_SECRET_FILE="${CLUSTER_PROFILE_DIR}/pull-secret"
+PULL_SECRET=$(cat ${CLUSTER_PULL_SECRET_FILE} ${LCA_PULL_SECRET_FILE} | jq -cs '.[0] * .[1]') # Merge the pull secrets to get everything we need
+BACKUP_SECRET_FILE="/var/run/ibu-backup-secret/.backup-secret"
+BACKUP_SECRET=$(jq -c . ${BACKUP_SECRET_FILE})
+remote_workdir=$(cat ${SHARED_DIR}/remote_workdir)
+instance_ip=$(cat ${SHARED_DIR}/public_address)
+host=$(cat ${SHARED_DIR}/ssh_user)
+ssh_host_ip="$host@$instance_ip"
+
+TARGET_VM_NAME="target"
+echo "${TARGET_VM_NAME}" > "${SHARED_DIR}/target_vm_name"
+
+SEED_VERSION=$(cat ${SHARED_DIR}/seed_version)
+SEED_IMAGE_TAG=$(cat ${SHARED_DIR}/seed_tag)
+SEED_IMAGE="${SEED_IMAGE}:${SEED_IMAGE_TAG}"
+
+# Save the pull secrets
+echo -n "${PULL_SECRET}" > ${SHARED_DIR}/.pull_secret.json
+echo -n "${BACKUP_SECRET}" > ${SHARED_DIR}/.backup_secret.json
+
+echo "Creating upgrade script..."
+cat <<EOF > ${SHARED_DIR}/image_based_install.sh
+#!/bin/bash
+set -euo pipefail
+
+export SEED_IMAGE="${SEED_IMAGE}"
+export SEED_VERSION="${SEED_VERSION}"
+export LCA_IMAGE="${LCA_PULL_REF}"
+export IBI_VM_NAME="${TARGET_VM_NAME}"
+
+cd ${remote_workdir}/ib-orchestrate-vm
+
+export REGISTRY_AUTH_FILE='${remote_workdir}/.pull_secret.json'
```

Comment:
This method of getting the pull secret to the ec2 instance is something I want to roll into the rest of the CI. This prevents the pull secrets from leaking in to the logs

Reply:
SCP is the preferred approach. The echo here is because we need a hybrid pull secret that can pull from CI repos as well as quay. It's probably worth revisiting (in another story) consolidating all pull secret generation for ibu/ibi to a single step that creates and uploads the pullsecret to the ec2 instance. (I'll add a story)